mender/templates/api-gateway-svc.yaml

```
@@ -29,13 +29,15 @@ spec:
   loadBalancerSourceRanges: {{- toYaml .Values.api_gateway.service.loadBalancerSourceRanges | nindent 4 }}
   {{- end }}
   ports:
+{{- if .Values.api_gateway.env.SSL }}
```

Shouldn't this be:
```suggestion
{{- if .Values.api_gateway.service.SSL }}
```
?

---

According to the current context, the SSL flag is defined in the env scope, thus the `api_gateway.env.SSL` variable name is correct.

However, changing the scope of that flag would be appropriate, since it affects the entire service and it is not localized to the environment of a particular container, although it has implications to such environment. 

In this case I suggest to rename the flag to something like this: `api_gateway.service.ssl_enabled` and group it together with the variables `api_gateway.service.httpsPort` and `api_gateway.service.httpsNodePort`.

Maybe @tranchitella or somebody else more familiarized with this context can bring more insight into how to proceed further.

Thanks.

---

@nucatus you are right, we should probably move the flat to the service. At this stage, though, keeping it as it is is the easiest choice and doesn't introduce a back-compatibility breaking change.